spacy/cli/debug_model.py

```diff
@@ -39,27 +41,46 @@ def debug_model_cli(
         "print_after_training": P2,
         "print_prediction": P3,
     }
-
-    if seed is not None:
-        msg.info(f"Fixing random seed: {seed}")
-        fix_random_seed(seed)
+    config_overrides = parse_config_overrides(ctx.args)
+    cfg = Config().from_disk(config_path)
+    with show_validation_error():
+        try:
+            _, config = util.load_model_from_config(cfg, overrides=config_overrides)
+        except ValueError as e:
+            msg.fail(str(e), exits=1)
+    use_gpu = config["training"]["use_gpu"]
     if use_gpu >= 0:
-        msg.info(f"Using GPU: {use_gpu}")
+        msg.info("Using GPU")
         require_gpu(use_gpu)
     else:
-        msg.info(f"Using CPU")
+        msg.info("Using CPU")
+    seed = config["pretraining"]["seed"]
+    if seed is not None:
+        msg.info(f"Fixing random seed: {seed}")
+        fix_random_seed(seed)
 
-    debug_model(
-        config_path, print_settings=print_settings,
-    )
+    component = config
+    parts = section.split(".")
+    for item in parts:
+        try:
```

Comment:
Ah, this is a practical solution! I was initially overthinking this in my head and was only thinking about more complex recursion 😅

We should consider moving this out to a helper function in util? I think it fits in well with our suite of dot notation <-> dict helpers and it totally seems like something that'll come up in different places. Like, selecting some property down the tree, checking if a config has a certain setting etc.